### ISSUE TITLE
TorXakis server fails to release port after #50

### ISSUE THREAD

**Issue by keremispirli** [TorXakis/TorXakis#55]
Looks like in case of a failure due to #50, TorXakis server releases the port only after its parent process exits. If parent process launches another TorXakis server without exiting, then the port is stuck as occupied by a non-existing process.

To reproduce, checkout 6e450ce8bc9a2c0b897aa54c5898de3cb96417f3 and run:
```
> %TXSREPO%\test\examps\LuckyPeople\TestLuckyPeople.bat %TXSREPO%
```

**Comment by keremispirli** [comment]
Looks like the reason is that cvc4 process is left open, occupying the ports (and log files!).
I will try with z3 to see if it behaves the same way.

**Comment by capitanbatata** [comment]
Since we got rid of the `.bat` files for testing, and since I'm working on launching the txsserver from the UI and handling exceptions properly, I'd propose to close this issue. What do you think @keremispirli?

**Comment by keremispirli** [comment]
Rather than the `.bat` files, the issue is that `cvc4.exe` cannot be properly closed in such cases. So in order to recreate this I ran sqatt with CVC4, killed two `txsui.exe`s during "HitAll7" test (one for Tester and one for Simulator) and observed whether `cvc4.exe` is killed properly. After the tests are finished i.e. sqatt exited and I'm back in terminal, I still see 2 `cvc4.exe` processes and two `txsserver.exe` processes alive. Killing `txsserver.exe` doesn't automatically make a `cvc4.exe` exit, either. But killing a `cvc4.exe` **does** make a `txsserver.exe` exit.

Same scenario with Z3 does not leave any `z3.exe` or `txsserver.exe` process running behind.

I also tried running all tests and killing all the `txsui.exe` processes that stay on the list long enough. With CVC4 some `cvc4.exe` and `txsserver.exe` processes are left running while others didn't. All the running `cvc4.exe` processes consume max CPU power, so it looks like the reason for cvc4 processes doesn't exit is that they are still busy solving the problem. Z3 doesn't leave any process behind. Assuming we don't treat Z3 and CVC4 differently, I think Z3:
 - either finishes it's calculations and processes the exit signal
 - or processes the exit signal even during calculations and quits calculation.

Anyway, it does look like this is because of CVC4's behaviour. If you think there's nothing we can do to force the solver process to exit, then I'm OK with `wontfix`.

Additionally, this experiment assumes that `txsui.exe` is somehow prematurely killed. If we're confident that such an incident is very unlikely to occur, then I'm also OK with `wontfix`.

**Comment by keremispirli** [comment]
@capitanbatata There's a TODO note at `createSMTEnv` function in SMTInternal.hs, that refers to #365 (bug 1954 in redmine) and asks if any changes are needed in that function. I think it's also related to this issue and you may want to take a look.
